diff -urN crystalspace-src-1.4.1/scripts/cs-config/cs-config.temppost crystalspace-src-1.4.1-patched/scripts/cs-config/cs-config.temppost
--- crystalspace-src-1.4.1/scripts/cs-config/cs-config.temppost	2011-12-17 05:33:20.000000000 +1100
+++ crystalspace-src-1.4.1-patched/scripts/cs-config/cs-config.temppost	2012-01-19 21:07:38.395439685 +1100
@@ -1,40 +1,14 @@
 # Detect type of the installation and the directories where the library files
 # reside in.
-libdir=""
 lflags=""
-
-for libname in lib${libcrystalspace}.so \
-	       lib${libcrystalspace}.dylib \
-	       lib${libcrystalspace}.a \
-	       ; do
-    if test -r "$prefix/Jamfile" && \
-	test -r "$prefix/$makeout/libs/$libname"; then
-	libdir="$prefix/$makeout/libs"
-	lflags="-L$libdir"
-    elif test -z "$libdir" && test -r "${exec_prefix}/lib/$libname"; then
-	libdir="${exec_prefix}/lib"
-	lflags="-L${exec_prefix}/lib"
-    else
-	my_IFS=$IFS; IFS=$PATH_SEPARATOR
-	for p in $_CRYSTAL_ACTUAL
-	  do
-	  if test -r "$p/Jamfile" && \
-	      test -r "$p/$makeout/libs/$libname"; then
-	      libdir="$p/$makeout/libs"
-	      lflags="-L$libdir"
-	      break
-	  elif test -z "$libdir" && test -r "${p}/lib/$libname"; then
-	      libdir="${p}/lib"
-	      lflags="-L$libdir"
-	      break
-	  fi
-	done
-	IFS=$my_IFS
-    fi
-    if test -n "$lflags" ; then
-        break
-    fi
-done
+arch=$(/bin/arch)
+if [ "$arch" = "x86_64" -o "$arch" = "s390x" -o "$arch" = "ppc64" \
+    -o "$arch" = "ia64" ];
+then
+    libdir="/usr/lib64"
+else
+    libdir="/usr/lib"
+fi
 
 # Look for build tools such as relaytool
 tools_sentinel=relaytool
@@ -288,17 +262,6 @@
   shift
 done
 
-# This is tested late in order to allow --help to work even if the Crystal
-# Space directory is not found.
-if test -z "$lflags"; then
-  cat 1>&2 <<EOF
-Failed to detect directory containing Crystal Space link libraries; aborting!
-Did you build Crystal Space? Did you set the CRYSTAL environment variable
-correctly?
-EOF
-  exit 1
-fi
-
 inc_dir_shown=""
 my_IFS=$IFS; IFS=$PATH_SEPARATOR
 for p in $includedir
diff -urN crystalspace-src-1.4.1/scripts/cs-config/Jamfile crystalspace-src-1.4.1-patched/scripts/cs-config/Jamfile
--- crystalspace-src-1.4.1/scripts/cs-config/Jamfile	2011-12-17 05:33:20.000000000 +1100
+++ crystalspace-src-1.4.1-patched/scripts/cs-config/Jamfile	2012-01-19 21:05:55.356440776 +1100
@@ -95,7 +95,7 @@
 fi
 
 CS_VERSION=$(PACKAGE_VERSION_SIGNIFICANT)
-CRYSTAL="\${CRYSTAL-$(prefix)\$PATH_SEPARATOR.}"
+CRYSTAL="\${CRYSTAL-$(prefix)}"
 CRYSTAL_$(ENV_VERSION)="\${CRYSTAL_$(ENV_VERSION)-\$CRYSTAL}"
 _CRYSTAL_ACTUAL="\$CRYSTAL_$(ENV_VERSION)"
 
@@ -127,7 +127,6 @@
 do
   prefix="\${p}"
   exec_prefix="\${prefix}"
-  makeout="$(LOCATE.OBJECTS)"
   version="$(PACKAGE_VERSION)"
   longversion="$(PACKAGE_STRING)"
   newincdir=""
